api/V1/recordingUtil.ts

```diff
@@ -79,23 +79,29 @@ export function latLngApproxDistance(a: [number, number], b: [number, number]):
   return part1 * R;
 }
 
-async function tryToMatchRecordingToStation(recording: Recording): Promise<StationId | null> {
+export async function tryToMatchRecordingToStation(recording: Recording, stations?: Station[]): Promise<Station | null> {
   // Match the recording to any stations that the group might have:
-  const group = await models.Group.getFromId(recording.GroupId);
-  const stations = await group.getStations();
+  if (!stations) {
```

Comment:
Recording when first uploaded may not have a GPS position I think.   If so we should just return straight away?